specs/src/tx_client.md

```diff
@@ -0,0 +1,273 @@
+- **Process**: Direct account usage -> PayForBlob creation -> Broadcast -> Confirmation
+- **Usage**: This bypasses having to spin up workers at all and can be used with a signle account for sequential submissions
+
+#### QueueBlob
+
+```go
+func (client *TxClient) QueueBlob(ctx context.Context, resultC chan SubmissionResult, blobs []*share.Blob, opts ...TxOption)
+```
+
+- **Path**: Direct job submission to worker queue
+- **Input**: Context, result channel, Celestia blobs, and transaction options
+- **Process**: Job queued directly -> Worker assignment -> Processing
+- **Usage**: This method adds a job into the queue directly, bypassing the normal submission flow
+
+### Submission API Summary
+
+All submission APIs ultimately delegate into the same broadcast and confirmation methods, they differ only in account selection, tx types and whether jobs go through the worker queue.
+
+## Broadcast
+
+Broadcasting steps are universal across SDK transactions and blob transactions.
+
+**Process:**
+
+1. **Build & Sign**: Transaction is built and signed using the Signer
+2. **Gas Estimation**:
+   - If no gas provided, simulate with minimal fee
+   - Call gas estimation service to determine gas usage and price
+   - If estimation fails due to a sequence mismatch, update local sequence to expected and retry
+3. **Fee Setting**: Fee is computed as `ceil(minGasPrice * gasLimit)` unless explicitly set by the user
+4. **Broadcast**: Transaction is encoded and submitted via gRPC:
+   - **Single-endpoint mode**: Retries on mismatch until success or failure
+   - **Multi-endpoint mode**: Broadcasts concurrently to multiple endpoints and accepts the first successful response
+5. **Tracker Entry**: On successful broadcast, a record is added to the transaction tracker containing (signer, sequence, txBytes, timestamp) so the transaction can be resubmitted or rolled back if needed
+6. **Sequence Increment**: After a successful broadcast, the local sequence for that account is incremented
+
+⚠️ **Note**: This approach assumes the TxClient is connected to a trusted consensus node in order to maintain replay protection.
+
+## Confirmation
+
+After broadcast, the TxClient continuously polls the chain for transaction status until resolution. This is the tx confrimation life cycle:
+
+- **Pending**: Transaction in the mempool but not yet included -> continue polling
+- **Committed**: Transaction included in a block
+  - If execution succeeded (code = OK): return success
+  - If execution failed: return an execution error and remove from tx tracker
+- **Rejected**:
+  - Transaction explicitly refused by the node during `ReCheck()`(This happens in the mempool after a block is committed).
+  - The client rolls back the sequence to the rejected transaction's sequence
+  - Removes the `TxTracker` entry
+  - Returns error
+- **Evicted**: Transaction dropped from mempool (e.g. low fees).
+  - The client resubmits using txBytes stored in the tracker
```

Comment:
do we talk about concept of TTL? We should probably specify that as it's the desired behaviour here as well (replay protection) ?

Reply:
We have not yet settled on what we want TTLs to look like. I'd only add them to the spec once we actually decide.